ape_solidity/compiler.py

```diff
@@ -270,68 +333,80 @@ def parse_contract_name(value: str) -> Tuple[Path, str]:
             input_contract_names: List[str] = []
             for contract_id in output.keys():
                 path, name = parse_contract_name(contract_id)
-                for input_file_path in files:
+                for input_file_path in input_file_paths:
                     if str(path) in str(input_file_path):
                         input_contract_names.append(name)
 
-            def load_dict(data: Union[str, dict]) -> Dict:
-                return data if isinstance(data, dict) else json.loads(data)
-
             for contract_name, contract_type in output.items():
                 contract_path, contract_name = parse_contract_name(contract_name)
-
                 if contract_name not in input_contract_names:
-                    # Contract was either not specified or was compiled
-                    # previously from a later-solidity version.
+                    # Only return ContractTypes explicitly asked for.
                     continue
 
-                contract_type["contractName"] = contract_name
-                contract_type["sourceId"] = (
-                    str(get_relative_path(base_path / contract_path, base_path))
-                    if base_path and contract_path.is_absolute()
-                    else str(contract_path)
-                )
-                contract_type["deploymentBytecode"] = {"bytecode": contract_type["bin"]}
-                contract_type["runtimeBytecode"] = {"bytecode": contract_type["bin-runtime"]}
-                contract_type["userdoc"] = load_dict(contract_type["userdoc"])
-                contract_type["devdoc"] = load_dict(contract_type["devdoc"])
+                deployment_bytecode = contract_type["bin"]
+                runtime_bytecode = contract_type["bin"]
+
+                # Skip library linking.
+                if "$__" in deployment_bytecode or "__$" in runtime_bytecode:
+                    logger.warning("Libraries must be deployed and configured separately.")
+                    continue
```

Comment:
Perfect, now just need config option for specifying libraries to pass through to `solc` and link

Reply:
One sneaky thing that can be done here I guess is that if you are on a local network and specify `auto`, have it automatically deploy any libraries using a test_account and add that to the config somehow.

Absolutely atrocious I know, but so is this `solc` feature

Reply:
I created a separate issue for that since this PR is getting out-of-hand

Reply:
We are at least able to compile `yearn-vaults` AND `chainlink-brownie-projects` now!